### ISSUE TITLE
[libepoxy:x64-windows] Python build failure

### ISSUE THREAD

**Issue by str4d** [microsoft/vcpkg#8860]
**Host Environment**
- OS: Windows 10 rev 1903

**To Reproduce**
```powershell
> git checkout 08e74979df124d9360343751cefcae615c2ec5ed
> .\vcpkg.exe install gtk:x64-windows
The following packages will be built and installed:
  * atk[core]:x64-windows
  * bzip2[core]:x64-windows
  * cairo[core]:x64-windows
  * dirent[core]:x64-windows
  * expat[core]:x64-windows
  * fontconfig[core]:x64-windows
  * freetype[core]:x64-windows
  * gdk-pixbuf[core]:x64-windows
  * gettext[core]:x64-windows
  * glib[core]:x64-windows
    gtk[core]:x64-windows
  * harfbuzz[core,glib,ucdn]:x64-windows
  * libepoxy[core]:x64-windows
  * libffi[core]:x64-windows
  * libiconv[core]:x64-windows
  * libpng[core]:x64-windows
  * pango[core]:x64-windows
  * pcre[core]:x64-windows
  * pixman[core]:x64-windows
  * ragel[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/21: libepoxy:x64-windows
Building package libepoxy[core]:x64-windows...
-- Using cached C:/Users/me/Documents/dev/git/vcpkg/downloads/anholt-libepoxy-1.5.3.tar.gz
-- Using source at C:/Users/me/Documents/dev/git/vcpkg/buildtrees/libepoxy/src/1.5.3-d7b9d32fcb
-- Configuring x64-windows-rel
CMake Error at C:/Users/me/Documents/dev/git/vcpkg/scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/python/python-3.7.3/python.exe C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/meson/meson-0.52.0/meson.py -Denable-glx=no -Denable-egl=no -Dcmake_prefix_path=C:/Users/me/Documents/dev/git/vcpkg/installed/x64-windows --buildtype plain --backend ninja --wrap-mode nodownload --default-library shared --prefix C:/Users/me/Documents/dev/git/vcpkg/packages/libepoxy_x64-windows C:/Users/me/Documents/dev/git/vcpkg/buildtrees/libepoxy/src/1.5.3-d7b9d32fcb
    Working Directory: C:/Users/me/Documents/dev/git/vcpkg/buildtrees/libepoxy/x64-windows-rel
    Error code: 2
    See logs for more information:
      C:\Users\me\Documents\dev\git\vcpkg\buildtrees\libepoxy\config-x64-windows-rel-out.log
      C:\Users\me\Documents\dev\git\vcpkg\buildtrees\libepoxy\config-x64-windows-rel-err.log

Call Stack (most recent call first):
  C:/Users/me/Documents/dev/git/vcpkg/scripts/cmake/vcpkg_configure_meson.cmake:58 (vcpkg_execute_required_process)
  C:/Users/me/Documents/dev/git/vcpkg/ports/libepoxy/portfile.cmake:14 (vcpkg_configure_meson)
  C:/Users/me/Documents/dev/git/vcpkg/scripts/ports.cmake:94 (include)


Error: Building package libepoxy:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libepoxy:x64-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.
```

**Failure logs** 

`buildtrees\libepoxy\config-x64-windows-rel-out.log`:
```
The Meson build system
Version: 0.52.0
Source dir: C:\Users\me\Documents\dev\git\vcpkg\buildtrees\libepoxy\src\1.5.3-d7b9d32fcb
Build dir: C:\Users\me\Documents\dev\git\vcpkg\buildtrees\libepoxy\x64-windows-rel
Build type: native build
WARNING: Unknown options: "enable-egl, enable-glx"
Project name: libepoxy
Project version: 1.5.3
Appending CFLAGS from environment: ' /DWIN32 /D_WINDOWS /W3 /utf-8  /MD /O2 /Gy /DNDEBUG /Z7'
Appending LDFLAGS from environment: ['/DEBUG', '/INCREMENTAL:NO', '/OPT:REF', '/OPT:ICF']
Appending CPPFLAGS from environment: ' '
Appending CFLAGS from environment: ' /DWIN32 /D_WINDOWS /W3 /utf-8  /MD /O2 /Gy /DNDEBUG /Z7'
Appending LDFLAGS from environment: ['/DEBUG', '/INCREMENTAL:NO', '/OPT:REF', '/OPT:ICF']
Appending CPPFLAGS from environment: ' '
C compiler for the host machine: cl (msvc 19.12.25831)
C linker for the host machine: link 19.12.25831
Host machine cpu family: x86_64
Host machine cpu: x86_64
Has header "KHR/khrplatform.h" : NO 
Compiler for C supports arguments -we4002: YES 
Compiler for C supports arguments -we4003: YES 
Compiler for C supports arguments -w14010: YES 
Compiler for C supports arguments -we4013: YES 
Compiler for C supports arguments -w14016: YES 
Compiler for C supports arguments -we4020: YES 
Compiler for C supports arguments -we4021: YES 
Compiler for C supports arguments -we4027: YES 
Compiler for C supports arguments -we4029: YES 
Compiler for C supports arguments -we4033: YES 
Compiler for C supports arguments -we4035: YES 
Compiler for C supports arguments -we4045: YES 
Compiler for C supports arguments -we4047: YES 
Compiler for C supports arguments -we4049: YES 
Compiler for C supports arguments -we4053: YES 
Compiler for C supports arguments -we4071: YES 
Compiler for C supports arguments -we4819: YES 
Library dl found: NO
Did not find pkg-config by name 'pkg-config'
Found Pkg-config: NO
Run-time dependency gl found: YES 
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency egl found: NO (tried pkgconfig and cmake)
Run-time dependency x11 found: NO (tried pkgconfig and cmake)
Run-time dependency glesv2 found: NO (tried pkgconfig and cmake)
Library libGLESv2 found: NO
Run-time dependency glesv1_cm found: NO (tried pkgconfig and cmake)
Library libGLESv1_CM found: NO
Library opengl32 found: YES
Library gdi32 found: YES
DEPRECATION: Project targetting '>= 0.47.0' but tried to use feature deprecated since '0.48.0': python3 module
Configuring config.h using configuration
Library dl found: NO
Checking for function "dlvsym" with dependency -ldl: NO 
Build targets in project: 13
WARNING: Deprecated features used:
 * 0.48.0: {'python3 module'}
Option buildtype is: plain [default: debugoptimized]
```

`buildtrees\libepoxy\config-x64-windows-rel-err.log`:
```
Traceback (most recent call last):
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\mesonmain.py", line 129, in run
    return options.run_func(options)
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\msetup.py", line 245, in run
    app.generate()
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\msetup.py", line 159, in generate
    self._generate(env)
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\msetup.py", line 215, in _generate
    intr.backend.generate(intr)
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\backend\ninjabackend.py", line 288, in generate
    self.ninja_command = environment.detect_ninja(log=True)
  File "C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\meson\meson-0.52.0\mesonbuild\environment.py", line 156, in detect_ninja
    name = os.path.basename(n)
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\ntpath.py", line 214, in basename
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\ntpath.py", line 183, in split
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Additional information**

I modified `downloads\tools\meson\meson-0.52.0\mesonbuild\environment.py` to print out `os.environ.get('PATH')` and confirmed that `C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\ninja\ninja-1.8.2` was present (it was at the end). I also confirmed that `ninja.exe` was accessible:
```powershell
> .\downloads\tools\python\python-3.7.3\python.exe
>>> import os
>>> os.access('downloads/tools/ninja/ninja-1.8.2/ninja.exe', os.F_OK | os.X_OK)
True
>>> import shutil
>>> print(shutil.which('ninja.exe', path='C:\\Users\\me\\Documents\\dev\\git\\vcpkg\\downloads\\tools\\ninja\\ninja-1.8.2;'))
C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\ninja\ninja-1.8.2\ninja.exe
>>> print(shutil.which('ninja', path='C:\\Users\\me\\Documents\\dev\\git\\vcpkg\\downloads\\tools\\ninja\\ninja-1.8.2;'))
None
```

**Comment by str4d** [comment]
Confirmed that I can get past this error with the following change to `downloads\tools\meson\meson-0.52.0\mesonbuild\environment.py`:
```diff
-    for n in [env_ninja] if env_ninja else ['ninja', 'ninja-build', 'samu']:
+    for n in [env_ninja] if env_ninja else ['ninja.exe', 'ninja-build', 'samu']:
```

I then run into the following:
```powershell
> .\vcpkg.exe install gtk:x64-windows
The following packages will be built and installed:
  * atk[core]:x64-windows
  * bzip2[core]:x64-windows
  * cairo[core]:x64-windows
  * dirent[core]:x64-windows
  * expat[core]:x64-windows
  * fontconfig[core]:x64-windows
  * freetype[core]:x64-windows
  * gdk-pixbuf[core]:x64-windows
  * gettext[core]:x64-windows
  * glib[core]:x64-windows
    gtk[core]:x64-windows
  * harfbuzz[core,glib,ucdn]:x64-windows
  * libepoxy[core]:x64-windows
  * libffi[core]:x64-windows
  * libiconv[core]:x64-windows
  * libpng[core]:x64-windows
  * pango[core]:x64-windows
  * pcre[core]:x64-windows
  * pixman[core]:x64-windows
  * ragel[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/21: libepoxy:x64-windows
Building package libepoxy[core]:x64-windows...
-- Using cached C:/Users/me/Documents/dev/git/vcpkg/downloads/anholt-libepoxy-1.5.3.tar.gz
-- Using source at C:/Users/me/Documents/dev/git/vcpkg/buildtrees/libepoxy/src/1.5.3-d7b9d32fcb
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Configuring x64-windows-dbg
-- Configuring x64-windows-dbg done
-- Package x64-windows-rel
-- Package x64-windows-dbg
-- Performing post-build validation
-- Performing post-build validation done
Building package libepoxy[core]:x64-windows... done
Installing package libepoxy[core]:x64-windows...
Installing package libepoxy[core]:x64-windows... done
Elapsed time for package libepoxy:x64-windows: 13.56 s
Starting package 2/21: zlib:x64-windows
Building package zlib[core]:x64-windows...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Extracting source C:/Users/me/Documents/dev/git/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb
-- Configuring x64-windows
CMake Error at C:/Users/me/Documents/dev/git/vcpkg/scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\me\Documents\dev\git\vcpkg\buildtrees\zlib\config-x64-windows-out.log

Call Stack (most recent call first):
  C:/Users/me/Documents/dev/git/vcpkg/scripts/cmake/vcpkg_configure_cmake.cmake:296 (vcpkg_execute_required_process)
  C:/Users/me/Documents/dev/git/vcpkg/ports/zlib/portfile.cmake:22 (vcpkg_configure_cmake)
  C:/Users/me/Documents/dev/git/vcpkg/scripts/ports.cmake:94 (include)


Error: Building package zlib:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x64-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by str4d** [comment]
`buildtrees\zlib\config-x64-windows-out.log`:
```
[1/2] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Users/me/Documents/dev/git/vcpkg/packages/zlib_x64-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Users/me/Documents/dev/git/vcpkg/packages/zlib_x64-windows""
-- The C compiler identification is unknown
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-rel/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe cmTC_c9fb3 && [1/2] Building C object CMakeFiles\cmTC_c9fb3.dir\testCCompiler.c.obj
    FAILED: CMakeFiles/cmTC_c9fb3.dir/testCCompiler.c.obj 
    C:\PROGRA~2\MICROS~3\2017\COMMUN~1\VC\Tools\MSVC\1412~1.258\bin\Hostx64\x64\cl.exe   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP -o CMakeFiles\cmTC_c9fb3.dir\testCCompiler.c.obj   -c testCCompiler.c
    Microsoft (R) C/C++ Optimizing Compiler Version 19.12.25831 for x64
    Copyright (C) Microsoft Corporation.  All rights reserved.
    
    cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
    testCCompiler.c
    c1: fatal error C1083: Cannot open source file: 'testCCompiler.c': No such file or directory
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)


-- Configuring incomplete, errors occurred!
See also "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-rel/CMakeFiles/CMakeOutput.log".
See also "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-rel/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Users/me/Documents/dev/git/vcpkg/packages/zlib_x64-windows/debug""
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb" "-DSKIP_INSTALL_FILES=ON" "-DSKIP_BUILD_EXAMPLES=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/me/Documents/dev/git/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DSKIP_INSTALL_HEADERS=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Users/me/Documents/dev/git/vcpkg/packages/zlib_x64-windows/debug""
-- The C compiler identification is unknown
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.12.25827/bin/Hostx64/x64/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-dbg/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Users/me/Documents/dev/git/vcpkg/downloads/tools/ninja/ninja-1.8.2/ninja.exe cmTC_5ff13 && [1/2] Building C object CMakeFiles\cmTC_5ff13.dir\testCCompiler.c.obj
    FAILED: CMakeFiles/cmTC_5ff13.dir/testCCompiler.c.obj 
    C:\PROGRA~2\MICROS~3\2017\COMMUN~1\VC\Tools\MSVC\1412~1.258\bin\Hostx64\x64\cl.exe   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP -o CMakeFiles\cmTC_5ff13.dir\testCCompiler.c.obj   -c testCCompiler.c
    Microsoft (R) C/C++ Optimizing Compiler Version 19.12.25831 for x64
    Copyright (C) Microsoft Corporation.  All rights reserved.
    
    cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
    testCCompiler.c
    c1: fatal error C1083: Cannot open source file: 'testCCompiler.c': No such file or directory
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)


-- Configuring incomplete, errors occurred!
See also "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/Users/me/Documents/dev/git/vcpkg/buildtrees/zlib/x64-windows-dbg/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
```

**Comment by Voskrese** [comment]
uninstall/delete env

`D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32`

**Comment by str4d** [comment]
Huh, that's weird. I don't have a `D:` drive...

**Comment by str4d** [comment]
Oh, I think that might be an artifact of the Python 3.7 distribution downloaded by the vcpkg tooling:
```powershell
vcpkg> .\downloads\tools\python\python-3.7.3\python.exe
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 21:26:53) [MSC v.1916 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import ntpath
>>> print(ntpath.__file__)
C:\Users\me\Documents\dev\git\vcpkg\downloads\tools\python\python-3.7.3\python37.zip\ntpath.pyc
```

Perhaps the `.pyc` file's line-number debug info contains the paths from the system that packaged it?

**Comment by Voskrese** [comment]
what about zlib ?
`-- The C compiler identification is unknown`

I think you need to install some add ons for the compiler
 and try again

at myself I do not observe such mistakes

**Comment by str4d** [comment]
The second error *might* be unrelated, but I do not know. I originally posted it because I saw it was inside the `ninja` command (which I'd just overridden in a single place per my comment), and thought the command might be failing for a related reason (it seems to be failing because it can't find what I assume are test program files being generated by `cmake`, which seems like it could be related to the first issue with `meson` finding `ninja`).

**Comment by Voskrese** [comment]
Most likely, you need to install components msvc

meson/python to zlib has nothing to do with this port

**Comment by str4d** [comment]
I was using Visual Studio Community 2017 (for which the 30-day trial had ended, but MSVC was still there). I uninstalled that, installed Visual Studio Community 2019 (turning off everything except "Desktop C++ development" during installation), and then deleted and re-downloaded the `vcpkg` git repo. This time it did not fail with the original Python / `ninja` issue. Closing as likely a weird local env thing.
